python/cuml/internals/device_support.py

```diff
@@ -0,0 +1,18 @@
+#
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+GPU_ENABLED = True
+CPU_ENABLED = True
```

Comment:
I initially liked this proposal, but as I started to implement it, I found it clunky in practice. Having to write a more complex expression than `if CPU_ENABLED:` to check support seems unnecessary, especially since these are really just build-time variables that are going to be populated. Is there an advantage to a flag here that outweighs the additional verbosity?